### PR TITLE
Battery list: show values directly from from system /Batteries

### LINKED ISSUE #1572: Battery detail page data issues
Most genset controllers provide a starter battery voltage reading. If it is available via `com.victronenergy.genset.*/StarterVoltage`, Venus OS seems to interpret the genset service as a battery. This is really helpful, as a low starter battery (voltage) could cause a genset start failure.

However, gui-v2 does not yet seem to be able to interpret the service correctly. In the overview of all batteries in the system (via tap on the SystemBattery in the Overview), no helpful values are given for the genset starter battery.

192 168 254 90_gui-v2_(Ekrano GX) (5)
192 168 254 90_gui-v2_(Ekrano GX) (6)

For comparison: On VRM, the service is always called “Generator Starter Battery” and shows the only available value, the voltage. I would expect something similar on gui-v2.

Bildschirmfoto 2024-10-11 um 18 29 56

### PR DESCRIPTION
**Pull request by blammit**
The /Batteries list provides values like "power", "voltage", and "soc", so these can be used to show the values in the battery list. This is better than assuming each /Batteries entry comes from a service that provides paths like /Power, /Voltage, /Soc, etc. and fetching the data values from those paths for each entry.

For example, the /Batteries list may contain a starter battery provided by a genset service, and in this case the voltage is provided by com.victronenergy.genset.*/StarterVoltage instead of /Voltage. The backend uses the /StarterVoltage to set the 'voltage' value for the entry in the /Batteries list, so the UI should show that value, instead of trying to find a /Voltage path that does not exist.

Also rework the UI layout for each delegate in the battery list, since the SOC should be vertically centered if the mode/discharging text is not available.

Fixes #1572

### CODE REVIEW

**Comment by DanielMcInnes** [comment]
Should we be seeing the leading separator?

Screenshot from 2024-10-23 12-15-48

**Comment by blammit** [comment]
@DanielMcInnes It's not in the design, but without it, it's difficult to make the "My starter battery" and the "0.03V" line up. We can fudge it but it won't really line up exactly anyway. The problem is that the design of having those multiple numbers in a row does not normally need to align horizontally with anything else.

**Comment by DanielMcInnes** [comment]
> @DanielMcInnes It's not in the design, but without it, it's difficult to make the "My starter battery" and the "0.03V" line up. We can fudge it but it won't really line up exactly anyway. The problem is that the design of having those multiple numbers in a row does not normally need to align horizontally with anything else.
> 
> So we can either fudge it or add the separators 🤔

fair enough

File: pages/battery/BatteryListPage.qml [hunk 1]
```diff
@@ -132,34 +173,65 @@ Page {
...
+		property real soc: NaN
+		property real temperature: NaN
+		property int timetogo: 0
+		property real voltage: NaN
+		readonly property int mode: Global.batteries.batteryMode(power)
+	}
+
 	Component {
-		id: batteryComponent
+	   id: batteryComponent
```
Comment (DanielMcInnes) [inline pages/battery/BatteryListPage.qml:231]: formatting

File: pages/battery/BatteryListPage.qml [hunk 2]
```diff
@@ -132,34 +173,65 @@ Page {
...
 
-		Battery {
-			id: battery
+		BatteryListDevice {
+		   id: battery
```
Comment (DanielMcInnes) [inline pages/battery/BatteryListPage.qml:234]: formatting

**Review by DanielMcInnes — APPROVED** [review]
lgtm, formatting nits

File: pages/battery/BatteryListPage.qml (hunk 2 above)
Comment (blammit) [reply pages/battery/BatteryListPage.qml:234]: Done

**Comment by Serjle** [comment]
@DanielMcInnes 
divider
Separator just between numbers

**Pull request merged**